Flaky test: TestController_watchAccountsChanges

[TestController_watchAccountsChanges: 5.0% (1 of 20 failed)](https://ci.status.im/job/status-go/job/tests-nightly/142/Reports/)

```
=== RUN   TestController_watchAccountsChanges
    controller_test.go:109: 
        	Error Trace:	/home/jenkins/workspace/status-go/tests-nightly/services/wallet/transfer/controller_test.go:109
        	Error:      	"[{ 0x0100000000000000000000000000000000000000000000000000000000000000 0x0000000000000000000000000000000000001234 1 0x0100000000000000000000000000000000000000000000000000000000000000 %!s(uint64=0) %!s(*types.Transaction=&{<nil> {0 0 <nil>} 0 {<nil>} {<nil>} {<nil>} {<nil>}}) %!s(bool=false) %!s(uint64=777) 0x0000000000000000000000000000000000000000 %!s(*types.Receipt=&{0 0 [] 0 0 [0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0] [] [0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0] [0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0] 0 <nil> <nil> [0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0] <nil> 0 <nil> <nil> <nil> <nil>}) %!s(*types.Log=&{[0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0] [] [] 0 [0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0] 0 [0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0] 0 false}) <nil> <nil>  %!s(transfer.MultiTransactionIDType=0)}]" should have 0 item(s), but has 1
        	Test:       	TestController_watchAccountsChanges
--- FAIL: TestController_watchAccountsChanges (0.27s)
```

https://ci.status.im/job/status-go/job/tests-nightly/142/testReport/github/com_status-im_status-go_services_wallet_transfer/TestController_watchAccountsChanges/

--- a/services/wallet/transfer/controller.go
+++ b/services/wallet/transfer/controller.go
@@ -150,8 +150,8 @@ func (c *Controller) startAccountWatcher(chainIDs []uint64) {
 		c.accWatcher = accountsevent.NewWatcher(c.accountsDB, c.accountFeed, func(changedAddresses []common.Address, eventType accountsevent.EventType, currentAddresses []common.Address) {
 			c.onAccountsChanged(changedAddresses, eventType, currentAddresses, chainIDs)
 		})
-		c.accWatcher.Start()
 	}
+	c.accWatcher.Start()
 }
 
 func (c *Controller) onAccountsChanged(changedAddresses []common.Address, eventType accountsevent.EventType, currentAddresses []common.Address, chainIDs []uint64) {

--- a/services/wallet/transfer/controller_test.go
+++ b/services/wallet/transfer/controller_test.go
@@ -80,12 +80,40 @@ func TestController_watchAccountsChanges(t *testing.T) {
 	require.NoError(t, err)
 	require.NotNil(t, ranges)
 
-	ch := make(chan accountsevent.Event)
-	// Subscribe for account changes
-	accountFeed.Subscribe(ch)
+	c.accWatcher = accountsevent.NewWatcher(c.accountsDB, c.accountFeed, func(changedAddresses []common.Address, eventType accountsevent.EventType, currentAddresses []common.Address) {
+		c.onAccountsChanged(changedAddresses, eventType, currentAddresses, []uint64{chainID})
+
+		// Quit channel event handler before  destroying the channel
+		go func() {
+			time.Sleep(1 * time.Millisecond)
+			// Wait for DB to be cleaned up
+			c.accWatcher.Stop()
+
+			// Check that transfers, blocks and block ranges were deleted
+			transfers, err := database.GetTransfersByAddress(chainID, address, big.NewInt(2), 1)
+			require.NoError(t, err)
+			require.Len(t, transfers, 0)
+
+			blocksDAO := &BlockDAO{walletDB}
+			block, err := blocksDAO.GetLastBlockByAddress(chainID, address, 1)
+			require.NoError(t, err)
+			require.Nil(t, block)
+
+			ranges, _, err = blockRangesDAO.getBlockRange(chainID, address)
+			require.NoError(t, err)
+			require.Nil(t, ranges.eth.FirstKnown)
+			require.Nil(t, ranges.eth.LastKnown)
+			require.Nil(t, ranges.eth.Start)
+			require.Nil(t, ranges.tokens.FirstKnown)
+			require.Nil(t, ranges.tokens.LastKnown)
+			require.Nil(t, ranges.tokens.Start)
+		}()
+	})
+	c.startAccountWatcher([]uint64{chainID})
 
 	// Watching accounts must start before sending event.
-	// To avoid running goroutine immediately, use any delay.
+	// To avoid running goroutine immediately and let the controller subscribe first,
+	// use any delay.
 	go func() {
 		time.Sleep(1 * time.Millisecond)
 
@@ -94,33 +122,6 @@ func TestController_watchAccountsChanges(t *testing.T) {
 			Accounts: []common.Address{address},
 		})
 	}()
-
-	c.startAccountWatcher([]uint64{chainID})
-
-	// Wait for event
-	<-ch
-
-	// Wait for DB to be cleaned up
-	c.accWatcher.Stop()
-
-	// Check that transfers, blocks and block ranges were deleted
-	transfers, err := database.GetTransfersByAddress(chainID, address, big.NewInt(2), 1)
-	require.NoError(t, err)
-	require.Len(t, transfers, 0)
-
-	blocksDAO := &BlockDAO{walletDB}
-	block, err := blocksDAO.GetLastBlockByAddress(chainID, address, 1)
-	require.NoError(t, err)
-	require.Nil(t, block)
-
-	ranges, _, err = blockRangesDAO.getBlockRange(chainID, address)
-	require.NoError(t, err)
-	require.Nil(t, ranges.eth.FirstKnown)
-	require.Nil(t, ranges.eth.LastKnown)
-	require.Nil(t, ranges.eth.Start)
-	require.Nil(t, ranges.tokens.FirstKnown)
-	require.Nil(t, ranges.tokens.LastKnown)
-	require.Nil(t, ranges.tokens.Start)
 }
 
 func TestController_cleanupAccountLeftovers(t *testing.T) {

